### PR TITLE
Gcndsp 324/spine services mock

### PR DESCRIPTION
**Pull request by awilkins**
Add spine services mock and more unit tests

The mock folder includes a Mountebank config for proxying the remote API and gathering more stubs.

To use this, configure it in the `docker-compose.yml` file, run the queries, visit http://localhost:2525 and the Imposters tab and copy new stubs into the main config.

### CODE REVIEW

File: acceptance/User.Specs/Hooks/Screenshot.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                    string Runname = title + DateTime.Now.ToString("yyyy-MM-dd-HH_mm_ss.jpg");
+
+                    string reportFolder = "./reports";
+                    var d = Directory.CreateDirectory(reportFolder);
+                    ss.SaveAsFile("./reports/" + Runname, ScreenshotImageFormat.Jpeg);
+                }
+            }
+            catch
+            {
+                //Console.WriteLine("catch");
```
Comment (leew) [inline acceptance/User.Specs/Hooks/Screenshot.cs]: Commented Code

File: acceptance/User.Specs/PageObjects/ApplicationLandingPageObject.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            _webDriver.Url = URL(PATH);
+        }
+
+        public bool IsStartButtonVisible()
+        {
+            var wait = new WebDriverWait(_webDriver, DefaultWait);
+            return wait.Until(driver => IsButtonVisible("Apply Now"));
+        }
+
+        // public void ClickStart(){
```
Comment (leew) [inline acceptance/User.Specs/PageObjects/ApplicationLandingPageObject.cs]: Commented Code

File: acceptance/User.Specs/Hooks/Screenshot.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+using TechTalk.SpecFlow;
+
+using OpenQA.Selenium;
+
+using GpConnect.DataSharing.User.Specs.Drivers;
+
+namespace GpConnect.DataSharing.User.Specs.Hooks
+{
+
```
Comment (leew) [inline acceptance/User.Specs/Hooks/Screenshot.cs]: Too much whitespace

File: acceptance/User.Specs/PageObjects/SoftwareSupplierPageObject.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+        public void SelectSupplier(string supplierName)
+        {
+            new SelectElement(SupplierList).SelectByText(supplierName);
+        }
+
+        private IWebElement ProductInputHint =>
+            _webDriver
+                .FindElement(By.Id("input-hint-softwaresupplierproduct"));
+
```
Comment (leew) [inline acceptance/User.Specs/PageObjects/SoftwareSupplierPageObject.cs]: Remove extra new line please

File: acceptance/User.Specs/Steps/ApplicationLandingStepDefinitions.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            _applicationLandingPage.Open();
+        }
+
+        [Then(@"A button to start the application is present")]
+        public void ThenAButtonToStartTheApplicationSearchIsPresent()
+        {
+            Assert.True(_applicationLandingPage.IsStartButtonVisible());
+        }
+
+        // [Then(@"the support email address is present")]
```
Comment (leew) [inline acceptance/User.Specs/Steps/ApplicationLandingStepDefinitions.cs]: please remove commented code

File: acceptance/User.Specs/Steps/TransparencyLandingStepDefinitions.cs [hunk 6]
```diff
@@ -34,23 +34,11 @@ public void ThenAButtonToStartTheTransparencySearchIsPresent()
...
-        {
-            Assert.True(_transparencyLandingPage.SupportPhone.Displayed);
-        }
-
-        [When(@"I click the start now button")]
-        public void WhenIClickTheStartNowButton()
-        {
-            _transparencyLandingPage.ClickStart();
-        }
+        // [When(@"I click the start now button")]
```
Comment (leew) [inline acceptance/User.Specs/Steps/TransparencyLandingStepDefinitions.cs]: Commented Code

File: deploy/Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+node {
+  try {
+
+    stage('Checkout') {
+      checkout scm
+    }
+
+    stage("Authenticate to AWS/k8s") {
+      // TODO : Do namespace here b/c we might need it for DB migration
```
Comment (leew) [inline deploy/Jenkinsfile]: This is TO-DONE

File: deploy/Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+
+      creds = awsCreds.tokenize(',\n')
+
+      withEnv(creds) {
+        sh 'rm -f ~/.kube/config'
+        sh 'aws eks --region eu-west-2 update-kubeconfig --name live-leks-cluster'
+      }
+
+
+      //   kubeconfig = sh(
```
Comment (leew) [inline deploy/Jenkinsfile]: Commented code

File: deploy/Jenkinsfile [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+          config = readFile(configFile)
+          config = replace(config)
+          writeFile(file: configFile, text: config)
+        }
+      }
+
+    }
+
+    stage('Database Migration') {
+      // TODO : Do a bit where we rename the job file
```
Comment (leew) [inline deploy/Jenkinsfile]: Fix or raise an issue please

File: deploy/Jenkinsfile [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+        }
+      }
+    }
+
+    stage('Deploy') {
+      k8sfiles = [
+        'namespace.yaml',
+        'deployment.yaml',
+        'service.yaml',
+        // 'ingress.yaml',
```
Comment (leew) [inline deploy/Jenkinsfile]: Uncomment code?

File: deploy/env/dev/database.tf [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+resource "aws_secretsmanager_secret" "database_password" {
+  name                    = "${local.prefix}/database-password"
+  recovery_window_in_days = 0
+  tags = {
+    Name = "${local.prefix}/database-password"
+  }
+}
+
+# MANUALLY store the database creds in the above secret in the following style
```
Comment (leew) [inline deploy/env/dev/database.tf]: I think that this has been raised as a seperate ticket

File: deploy/env/dev/security.tf [hunk 12]
```diff
@@ -0,0 +1,61 @@
+# K8s worker security group
+# TODO : migrate to Security Groups for Pods
```
Comment (leew) [inline deploy/env/dev/security.tf]: Resolve or another ticket

File: deploy/env/dev/store.tf [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+#   length = 20
+#   special = true
+# }
+
+# resource "aws_ssm_parameter" "database_password_parameter" {
+#   name        = "${local.prefix}-rds-master-password"
+#   description = "RDS database password"
+#   type        = "SecureString"
+#   value       = random_password.default.result
+# }
```
Comment (leew) [inline deploy/env/dev/store.tf]: Commented Code

File: k8s/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+            - name: ASPNETCORE_URLS
+              value: http://*:80
+            - name: SiteApi__BaseUrl
+              value: http://api
+            - name: SupplierApi__BaseUrl
+              value: http://api
+            - name: AgreementApi__BaseUrl
+              value: http://api
+            - name: OrganisationApi__BaseUrl
+              value: https://uat.directory.spineservices.nhs.uk
```
Comment (leew) [inline k8s/deployment.yaml:59]: We need to be replaced with an environment specific URL

**Comment by awilkins** [comment]
This PR is now too large. Will re-propose in chunks.

File: modules/mocks/start.sh [hunk 15]
```diff
@@ -1 +0,0 @@
-docker run -p 5555:5555 -v ${PWD}:/tmp/config bbyars/mountebank mb --configfile /tmp/config/mountebank.json
```
Comment (leew) [inline modules/mocks/start.sh]: Please restore this file it is helpful for local development, or please suggest an alternative approach

**Review by leew — COMMENTED** [review]
There are lots of changes mixed up in this PR. It has got bloated. Either we try and pick it apart to be able to prove each element in isolation or we accept the whole lot and might need to do some fast fixing tomorrow.

**Pull request closed without merging**